doc/rst/language/spec/iterators.rst

```diff
@@ -127,10 +127,12 @@ iteration, the iterator yields a value and the body is executed.
 Iterators as Arrays
 ~~~~~~~~~~~~~~~~~~~
 
-If an iterator function is captured into a new variable declaration or
-assigned to an array, the iterator is iterated over in total and the
-expression evaluates to a one-dimensional arithmetic array that contains
-the values returned by the iterator on each iteration.
+If an iterator function is captured into a new variable declaration,
+passed to an `in` intent argument, or assigned to an array, the iterator
+is iterated over in total and the expression evaluates to a rectangular
+array that contains the values returned by the iterator on each
+iteration. Iterators can pass to `in` intent arguments declared with a
+compatible array type expression.
```

Comment:
How does this address the case where the variable or argument has a declared type?

In some of these cases, at least in the case of "assigned to an array", a zippered iteration is used. This should be explained.

"Iterators **_can be passed_** to \`in\` intent" ?

Reply:
@vasslitvinov  - I've rewritten this paragraph. Please have another look.

Reply:
It is meant to include new variables declared with array type

Reply:
Then, how about stating that explicitly?

"Capturing an iterator into a new variable declared without an explicit type or declared with an array type" ?

We probably want to go further and restrict it to "with a rectangular array type" since we say later in this sentence that the created array is rectangular.

Reply:
I added a sentence, does it address your concern?